### PR TITLE
PBM-661: CI tests

### PR DESCRIPTION
**Pull request by dAdAbird**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | failed |
| mongodb 4.0 | failed |
| mongodb 4.2 | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | failed |
| mongodb 4.0 | failed |
| mongodb 4.2 | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | passed |
| mongodb 4.0 | passed |
| mongodb 4.2 | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | failed |
| mongodb 4.0 | failed |
| mongodb 4.2 | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | failed |
| mongodb 4.0 | failed |
| mongodb 4.2 | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| Restore on new cluster mongodb 3.6 | failed |
| Restore on new cluster mongodb 4.0 | failed |
| Restore on new cluster mongodb 4.2 | failed |
| Sharded cluster mongodb 3.6 | failed |
| Sharded cluster mongodb 4.0 | failed |
| Sharded cluster mongodb 4.2 | failed |
| Non-sharded replicaset mongodb 3.6 | failed |
| Non-sharded replicaset mongodb 4.0 | failed |
| Non-sharded replicaset mongodb 4.2 | failed |

File: Jenkinsfile [hunk 1]
```diff
@@ -12,6 +12,42 @@ void makeReport() {
     }
 }
 
+void runTest(String TEST_NAME, String TEST_SCRIPT, String MONGO_VERSION) {
+    def mkey = "$TEST_NAME mongodb $MONGO_VERSION"
+
+    testsReportMap[mkey] = 'failed'
+
+    sh '''
```
Comment (hors) [inline Jenkinsfile:20]:
```suggestion
    sh """
```

File: Jenkinsfile [hunk 2]
```diff
@@ -12,6 +12,42 @@ void makeReport() {
...
+        export MONGODB_VERSION=$MONGO_VERSION
+        export PBM_TESTS_NO_BUILD=true
+        ./e2e-tests/$TEST_SCRIPT
+    '''
```
Comment (hors) [inline Jenkinsfile:24]:
```suggestion
    """
```

File: Jenkinsfile [hunk 3]
```diff
@@ -12,6 +12,42 @@ void makeReport() {
...
+
+    testsReportMap[mkey] = 'passed'
+}
+
+void prepareCluster(String CLUSTER_POSTFIX) {
+    withCredentials([file(credentialsId: 'PBM-AWS-S3', variable: 'PBM_AWS_S3_YML'), file(credentialsId: 'PBM-GCS-S3', variable: 'PBM_GCS_S3_YML')]) {
+        sh '''
```
Comment (hors) [inline Jenkinsfile:31]:
```suggestion
        sh """
```

File: Jenkinsfile [hunk 4]
```diff
@@ -12,6 +12,42 @@ void makeReport() {
...
+            sed -i s:pbme2etest:pbme2etest-${CLUSTER_POSTFIX}:g ./e2e-tests/docker/conf/gcs.yaml
+
+            chmod 664 ./e2e-tests/docker/conf/aws.yaml
+            chmod 664 ./e2e-tests/docker/conf/gcs.yaml
+
+            docker-compose -f ./e2e-tests/docker/docker-compose.yaml build
+            openssl rand -base64 756 > ./e2e-tests/docker/keyFile
+            sudo chown 1001:1001 ./e2e-tests/docker/keyFile
+            sudo chmod 400 ./e2e-tests/docker/keyFile
+        '''
```
Comment (hors) [inline Jenkinsfile:47]:
```suggestion
        """
```

**Review by hors — CHANGES_REQUESTED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| Restore on new cluster mongodb 3.6 | failed |
| Restore on new cluster mongodb 4.0 | failed |
| Restore on new cluster mongodb 4.2 | failed |
| Sharded cluster mongodb 4.0 | failed |
| Non-sharded replicaset mongodb 4.0 | failed |
| Sharded cluster mongodb 3.6 | passed |
| Non-sharded replicaset mongodb 4.2 | failed |
| Sharded cluster mongodb 4.2 | failed |
| Non-sharded replicaset mongodb 3.6 | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| Restore on new cluster mongodb 3.6 | failed |
| Restore on new cluster mongodb 4.0 | failed |
| Restore on new cluster mongodb 4.2 | failed |
| Sharded cluster mongodb 3.6 | passed |
| Sharded cluster mongodb 4.0 | passed |
| Sharded cluster mongodb 4.2 | failed |
| Non-sharded replicaset mongodb 3.6 | passed |
| Non-sharded replicaset mongodb 4.0 | failed |
| Non-sharded replicaset mongodb 4.2 | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| Restore on new cluster mongodb 3.6 | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| Restore on new cluster mongodb 3.6 | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| Restore on new cluster mongodb 3.6 | passed |
| Restore on new cluster mongodb 4.0 | passed |
| Restore on new cluster mongodb 4.2 | passed |
| Sharded cluster mongodb 4.0 | failed |
| Sharded cluster mongodb 3.6 | passed |
| Non-sharded replicaset mongodb 3.6 | passed |
| Non-sharded replicaset mongodb 4.0 | passed |
| Sharded cluster mongodb 4.2 | passed |
| Non-sharded replicaset mongodb 4.2 | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| Restore on new cluster mongodb 3.6 | passed |
| Restore on new cluster mongodb 4.2 | passed |
| Non-sharded replicaset mongodb 3.6 | passed |
| Restore on new cluster mongodb 4.0 | passed |
| Sharded cluster mongodb 4.2 | passed |
| Sharded cluster mongodb 3.6 | passed |
| Sharded cluster mongodb 4.0 | passed |
| Non-sharded replicaset mongodb 4.0 | failed |
| Non-sharded replicaset mongodb 4.2 | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| Restore on new cluster mongodb 3.6 | passed |
| Restore on new cluster mongodb 4.0 | passed |
| Sharded cluster mongodb 3.6 | failed |
| Restore on new cluster mongodb 4.2 | passed |
| Sharded cluster mongodb 4.0 | failed |
| Non-sharded replicaset mongodb 3.6 | failed |
| Sharded cluster mongodb 4.2 | failed |
| Non-sharded replicaset mongodb 4.0 | failed |
| Non-sharded replicaset mongodb 4.2 | failed |

**Review by akira-kurogane — APPROVED** [review]
LGTM. Thanks for the CI testing TLC.

**Review by hors — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| Restore on new cluster mongodb 3.6 | passed |
| Restore on new cluster mongodb 4.0 | passed |
| Restore on new cluster mongodb 4.2 | passed |
| Sharded cluster mongodb 3.6 | failed |
| Non-sharded replicaset mongodb 3.6 | passed |
| Sharded cluster mongodb 4.0 | passed |
| Sharded cluster mongodb 4.2 | passed |
| Non-sharded replicaset mongodb 4.0 | passed |
| Non-sharded replicaset mongodb 4.2 | passed |

**Pull request merged**